### ISSUE TITLE
[research] Website perf should be excellent!

### ISSUE THREAD

**Issue by olizilla** [web3-storage/web3.storage#94]
Some things to investigate:
- Why is `magic-sdk` pulling in 3MB of resources from https://assets.auth.magic.link across 19 requests
- We need to trim down the 3MB of tailwind css.
- #61
- https://github.com/web3-storage/web3.storage/issues/195
- Logging in using Github takes a very long time with little feedback.

**Comment by olizilla** [comment]
**Requests to assets.auth.magic.link on staging** 
<img width="1504" alt="Screenshot 2021-07-15 at 14 56 23" src="https://user-images.githubusercontent.com/58871/125800218-5ea8ec44-ea44-4064-877a-6199ed426803.png">

**Requests to assets.auth.magic.link on nft.storage**
<img width="1504" alt="Screenshot 2021-07-15 at 14 51 32" src="https://user-images.githubusercontent.com/58871/125800104-9b952da7-cf70-42bf-ac9a-eba79f747cc3.png">

**Comment by olizilla** [comment]
@jdiogopeixoto can you get the css-purge / tailwind optimiser working on the website build to trim down that 3MB css bundle please.

**Comment by jdiogopeixoto** [comment]
@olizilla added `purge` to the #97

**Comment by alanshaw** [comment]
I can't see any documentation for reducing those requests from magic 😭

**Comment by drewdelano** [comment]
> Why is magic-sdk pulling in 3MB of resources from https://assets.auth.magic.link/ across 19 requests

I don't think we have any control over this because it's an external provider.  On slightly optimistic news, it looks like the amount of transferred data is only around 740kb after gzip does it's thing.

> Logging in using Github takes a very long time with little feedback.

I think this is just a problem in OAuth land. It currently takes about 7 seconds which is a little slower than average

**Comment by JeffLowe** [comment]
@drewdelano 
Continue perf review of site.  As you find any areas of concern, let's ticket separately (using this ticket as the research tracking ticket we can close once review is done).

**Comment by drewdelano** [comment]
I continued to look, but everything seems to be working well now as far as I can tell

**Comment by olizilla** [comment]
Please back this up with metrics. Let's track our lighthouse score, or any similar prefered measure that we can record in CI and ensure that we don't introduce regressions and work towards improving.

**Comment by drewdelano** [comment]
[web3.storage-20220621T094616.zip](https://github.com/web3-storage/web3.storage/files/8950862/web3.storage-20220621T094616.zip)
![image](https://user-images.githubusercontent.com/22156330/174854864-6399f40e-3331-45e1-b848-a9663add7a26.png)

**Comment by drewdelano** [comment]
> that we can record in CI and ensure that we don't introduce regressions and work towards improving.

Does this ticket include adding this to the CI pipeline or just taking a baseline?

**Comment by olizilla** [comment]
This ticket is a provocation to make our web perf excellent, and keep it excellent. It was deliberately open ended so that whoever picks it up gets to figure out a neat way of making that happen. Suggestions welcome!

In geneal, please approact all these issues with a "how can I leave this better than I found it?". Round trip times on a fully async team can be really slow. You can avoid getting blocked by making proposals and PRs to make the project better! There gonna be welcomed! Even if we need to change direction, the push the conversation forward more quickly.

**Comment by JeffLowe** [comment]
FYI large performance concern related to rendering discovered and getting corrected in #1953 
That will have a significant impact on performance across the site.
